db/database.go

```diff
@@ -1380,6 +1380,25 @@ func (context *DatabaseContext) ObtainManagementEndpoints() ([]string, error) {
 	return base.GoCBBucketMgmtEndpoints(gocbBucket.Bucket)
 }
 
+func (context *DatabaseContext) ObtainManagementEndpointsAndHTTPClient() ([]string, *http.Client, error) {
+	gocbBucket, ok := base.AsGoCBBucket(context.Bucket)
+	if !ok {
+		base.Warnf("Database %v: Unable to get server management endpoints. Underlying bucket type was not GoCBBucket.", base.MD(context.Name))
+		return nil, nil, nil
+	}
+
+	endpoints, err := base.GoCBBucketMgmtEndpoints(gocbBucket.Bucket)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	httpClient := &http.Client{
+		Transport: gocbBucket.Bucket.IoRouter().HttpClient().Transport,
```

Comment:
I think this is going to return a mutable reference to that bucket's transport?  Do we want to make a copy instead, to avoid callers of this method inadvertently modifying the underlying bucket client's transport?

Reply:
Yeah, thats true. Can utilize the http.Transport Clone() method here.